wordpress-mobile/WordPress-Android
# Adds "Register Domain" CTA if the site has an unused domain credit

Fixes #9815 Subtask: Create a Call to Action on My Site screen

This PR adds the ability to register a domain if there is an available domain credit. On the **My Site** screen, it will check the current plan for the selected site to see if there is any available domain credit. If there is a domain credit, it will show a call to action which allows the user to redeem that domain credit.

I decided not to show an error message to the user if the plans fail to load. My assumption is that in most cases there is no domain credit, so it would be better to just not show the error if the plans fail to load rather than show an unexpected error message that might confuse the user. I'd be happy to change if this does not make sense.

To test:
1. Install the wasabi build.
2. Switch to a site on a business plan with unclaimed domain credit 
3. The CTA should be displayed near the top of the screen, click on "Register Domain"
4. Pick a domain (preferably .blog).
5. Tap on "CHOOSE DOMAIN" and navigate to contact form details.
6. Fill out the your contact details (if they haven't already been prefilled)
7. Click "REGISTER DOMAIN"
8. If successful, the **Success** screen should be displayed. Click on "NEXT". You should now be returned to the **My SIte** screen

PS: Follow domain testing guideline and make sure if you purchase a domain to cancel it.

![Screenshots](https://user-images.githubusercontent.com/897223/57574107-2022af00-73f0-11e9-9224-d308c703e32a.png)
